server/src/main/java/io/spine/server/entity/storage/EntityColumnCache.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2018, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.entity.storage;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.spine.annotation.Internal;
+import io.spine.server.entity.Entity;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.server.entity.storage.Columns.obtainColumns;
+import static io.spine.validate.Validate.checkNotEmptyOrBlank;
+import static java.lang.String.format;
+import static java.util.Collections.synchronizedMap;
+
+/**
+ * A class designated to store cached {@linkplain EntityColumn entity column metadata} for a single
+ * {@link Entity} class.
+ *
+ * <p>The cache remains empty on creation. The {@linkplain EntityColumn column metadata} is retrieved
+ * on the first access to the {@linkplain EntityColumnCache cache} instance and then is stored
```

Comment:
`and stored in it`. There is no need to mention the implementation details here.

I would even remove the references to `Columns` usage in the paragraph below (or the whole paragraph).